MonoGame.Framework/Platform/Native/MessageBox.Native.cs

```diff
@@ -8,13 +8,24 @@
 
 namespace Microsoft.Xna.Framework.Input;
 
+
 public static partial class MessageBox
 {
     internal static unsafe MGP_Window* _window;
 
     private static unsafe Task<int?> PlatformShow(string title, string description, List<string> buttons)
     {
-        var result = MGP.Window_ShowMessageBox(_window, title, description, buttons.ToArray(), buttons.Count);
+        var button_bytes = new List<nint>();
+
+        byte* _title = stackalloc byte[StringInterop.GetMaxSize(title)];
+        StringInterop.CopyString(_title, title);
+        byte* _description = stackalloc byte[StringInterop.GetMaxSize(description)];
+        StringInterop.CopyString(_description, description);
+        byte* _buttons = stackalloc byte[StringInterop.GetMaxSize(buttons)];
+        StringInterop.CopyStrings(_buttons, buttons);
```

Comment:
We convert the list of strings to one string that is null delimited.  Seems like a easier way to interop this with native code and has low chances of issues.